src/LowerTriangularArrays/lower_triangular_array.jl

```diff
@@ -130,95 +141,138 @@ for f in (:zeros, :ones, :rand, :randn)
     @eval begin
         # use ArrayType from LowerTriangularArray parameter
         function Base.$f(
-            ::Type{LowerTriangularArray{T, N, ArrayType}}, 
-            m::Integer,
-            n::Integer,
+            ::Type{LowerTriangularArray{T, N, ArrayType, S}}, 
+            lmax::Integer,
+            mmax::Integer,
+            I::Vararg{Integer, M},
+        ) where {T, N, M, ArrayType, S<:AbstractSpectrum}
+            ArrayType_ = nonparametric_type(ArrayType)
+            return LowerTriangularArray(ArrayType_($f(T, nonzeros(lmax, mmax), I...)), Spectrum(lmax, mmax))
+        end
+
+        function Base.$f(
+            ::Type{LowerTriangularArray{T, N, ArrayType, S}}, 
+            spectrum::AbstractSpectrum,
             I::Vararg{Integer, M},
-        ) where {T, N, M, ArrayType}
+        ) where {T, N, M, ArrayType, S<:AbstractSpectrum}
             ArrayType_ = nonparametric_type(ArrayType)
-            return LowerTriangularArray(ArrayType_($f(T, nonzeros(m, n), I...)), m, n)
+            return LowerTriangularArray(ArrayType_($f(T, nonzeros(spectrum), I...)), spectrum)
         end
         
         # default CPU, use Array
         function Base.$f(
             ::Type{LowerTriangularArray{T}},
-            m::Integer,
-            n::Integer, 
+            lmax::Integer,
+            mmax::Integer, 
             I::Vararg{Integer, M},
         ) where {T, M}
-            return LowerTriangularArray($f(T, nonzeros(m, n), I...), m, n)
+            return LowerTriangularArray($f(T, nonzeros(lmax, mmax), I...), Spectrum(lmax, mmax))
+        end
+
+        function Base.$f(
+            ::Type{LowerTriangularMatrix{T}},
+            lmax::Integer,
+            mmax::Integer, 
+        ) where T
+            return LowerTriangularMatrix($f(T, nonzeros(lmax, mmax)), Spectrum(lmax, mmax))
         end
         
+        function Base.$f(
+            ::Type{LowerTriangularArray{T}},
+            spectrum::AbstractSpectrum,
+            I::Vararg{Integer, M},
+        ) where {T, M}
+            return LowerTriangularArray($f(T, nonzeros(spectrum), I...), spectrum)
+        end
+
+        function Base.$f(
+            ::Type{LowerTriangularMatrix{T}},
+            spectrum::AbstractSpectrum,
```

Comment:
`spectrum` implies a `LowerTriangularArray` so could be `zeros(spectrum)` too

Reply:
That's defined just a few lines below :)